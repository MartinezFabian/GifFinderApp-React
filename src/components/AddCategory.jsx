import { useState } from 'react';
import styles from './AddCategory.module.css';
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('');

  const onInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const onFormSubmit = (e) => {
    e.preventDefault();

    const value = inputValue.trim().toLowerCase();

    if (value.length <= 1) return;

    onNewCategory(value);

    setInputValue('');
  };

  return (
    <form onSubmit={onFormSubmit} className={styles.form}>
      <input
        type="text"
        placeholder="Buscar Gif"
        onChange={onInputChange}
        value={inputValue}
        className={styles.form__input}
      />
      <button type="submit" onSubmit={onFormSubmit} className={styles.form__button}>
        <svg
          xmlns="http://www.w3.org/2000/svg"
          fill="none"
          viewBox="0 0 24 24"
          strokeWidth={1.5}
          stroke="white"
          className={styles.form__icon}
        >
          <path
            strokeLinecap="round"
            strokeLinejoin="round"
            d="M21 21l-5.197-5.197m0 0A7.5 7.5 0 105.196 5.196a7.5 7.5 0 0010.607 10.607z"
          />
        </svg>
      </button>
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
