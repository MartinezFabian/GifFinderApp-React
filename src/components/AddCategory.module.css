.form {
  width: 100%;
  display: flex;
}

@media (min-width: 468px) {
  .form {
    width: 42rem;
  }
}

.form__input {
  width: 80%;
  padding: 0.5rem 2rem;
  border-radius: 2rem 0 0 2rem;
}

.form__button {
  width: 20%;
  background-color: var(--main--color);
  border: none;
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 0 2rem 2rem 0;
}

.form__button:hover {
  cursor: pointer;
}

.form__icon {
  color: var(--white);
  width: 60%;
  stroke: var(--white);
}
