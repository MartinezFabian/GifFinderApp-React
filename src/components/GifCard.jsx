import styles from './GifCard.module.css';

export const GifCard = ({ id, title, url }) => {
  return (
    <article className={styles.card}>
      <img src={url} alt={title} className={styles.card__img} />
      <h3 className={styles.card__title}>{title}</h3>
    </article>
  );
};
