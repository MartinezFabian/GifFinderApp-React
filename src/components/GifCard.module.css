.card {
  background-color: var(--card-color);
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  gap: 2rem;
  margin-bottom: 4rem;
  border-radius: 1rem;
  overflow: hidden;
  box-shadow: rgba(50, 50, 93, 0.25) 0px 6px 12px -2px, rgba(0, 0, 0, 0.3) 0px 3px 7px -3px;
  transition: transform 0.5s ease-out;
}

.card:hover {
  transform: scale(1.05);
}

@media (min-width: 468px) {
  .card {
    margin-bottom: unset;
  }
}

.card__img {
  height: 22rem;
  object-fit: cover;
}

.card__title {
  text-align: center;
}
