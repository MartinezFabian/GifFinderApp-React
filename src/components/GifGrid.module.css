.category {
  margin-top: 8rem;
}

.category__heading {
  text-align: center;
  text-transform: uppercase;
  font-size: 3rem;
  margin-bottom: 6rem;
}

@media (min-width: 468px) {
  .category__grid {
    display: grid;
    grid-template-columns: repeat(auto-fill, 27rem);
    gap: 4rem;
    justify-content: center;
  }
}
