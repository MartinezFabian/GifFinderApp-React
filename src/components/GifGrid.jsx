import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifCard } from './GifCard';
import styles from './GifGrid.module.css';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {
  const { gifs, isLoading } = useFetchGifs(category);

  return (
    <section className={styles.category}>
      <h2 className={styles.category__heading}>{category}</h2>

      {isLoading ? (
        <div className={styles.category__loader}>
          <span data-testid="loader" className={styles.loader}></span>
        </div>
      ) : null}

      <div className={styles.category__grid}>
        {gifs.map(({ id, title, url }) => (
          <GifCard key={id} title={title} url={url}></GifCard>
        ))}
      </div>
    </section>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
