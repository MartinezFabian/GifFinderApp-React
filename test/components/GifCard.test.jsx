import { render, screen } from '@testing-library/react';
import { GifCard } from '../../src/components/GifCard';

/* eslint-disable no-undef */
describe('tests in <GifCard />', () => {
  const title = 'messi';
  const url = 'https://messi.com/messi.jpg';

  test('must match the snapshot', () => {
    const { container } = render(<GifCard title={title} url={url}></GifCard>);

    expect(container).toMatchSnapshot();
  });

  test('the image must show the url and the alt', () => {
    render(<GifCard title={title} url={url}></GifCard>);

    const { src, alt } = screen.getByRole('img');

    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('must show the title', () => {
    render(<GifCard title={title} url={url}></GifCard>);

    expect(screen.getByText(title)).toBeTruthy();
  });
});
