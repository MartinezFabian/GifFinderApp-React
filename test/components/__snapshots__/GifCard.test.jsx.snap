// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in <GifCard /> must match the snapshot 1`] = `
<div>
  <article
    class="card"
  >
    <img
      alt="messi"
      class="card__img"
      src="https://messi.com/messi.jpg"
    />
    <h3
      class="card__title"
    >
      messi
    </h3>
  </article>
</div>
`;
